analyzer/codechecker_analyzer/analyzers/gcc/analyzer.py

```diff
@@ -0,0 +1,274 @@
+            with open(original_env_file, 'rb') as env_file:
+                env = pickle.load(env_file, encoding='utf-8')
+
+        return super().analyze(analyzer_cmd, res_handler, proc_callback, env)
+
+    def post_analyze(self, result_handler: GccResultHandler):
+        """
+        Post process the reuslts after the analysis.
+        Will copy the sarif files created by gcc into the root of the reports
+        folder.
+        Renames the source plist files to *.plist.bak because
+        The report parsing of the Parse command is done recursively.
+
+        """
+        pass
+
+    @classmethod
+    def resolve_missing_binary(cls, configured_binary, env):
+        """
+        In case of the configured binary for the analyzer is not found in the
+        PATH, this method is used to find a callable binary.
+        """
+
+        LOG.error("%s not found in path for GCC!", configured_binary)
+
+        # if os.path.isabs(configured_binary):
+        #     # Do not autoresolve if the path is an absolute path as there
+        #     # is nothing we could auto-resolve that way.
+        #     return False
+
+        # cppcheck = get_binary_in_path(['g++-13'],
+        #                               r'^cppcheck(-\d+(\.\d+){0,2})?$',
+        #                               env)
+
+        # if cppcheck:
+        #     LOG.debug("Using '%s' for Cppcheck!", cppcheck)
+        # return cppcheck
+
+    @classmethod
+    def __get_analyzer_version(cls, analyzer_binary, env):
+        """
+        Return the analyzer version.
+        """
+        # --version outputs a lot of garbage as well (like copyright info),
+        # this only contains the version info.
+        version = [analyzer_binary, '-dumpfullversion']
+        try:
+            output = subprocess.check_output(version,
+                                             env=env,
+                                             encoding="utf-8",
+                                             errors="ignore")
+            return output
+        except (subprocess.CalledProcessError, OSError) as oerr:
+            LOG.warning("Failed to get analyzer version: %s",
+                        ' '.join(version))
+            LOG.warning(oerr)
+
+        return None
+
+    @classmethod
+    def version_compatible(cls, configured_binary, environ):
+        """
+        Check the version compatibility of the given analyzer binary.
+        """
+        analyzer_version = \
+            cls.__get_analyzer_version(configured_binary, environ)
+
+        # The analyzer version should be above 13.0.0 because the
+        # '-fdiagnostics-format=sarif-file' argument was introduced in this
+        # release.
+        if analyzer_version >= StrictVersion("13.0.0"):
+            return True
+
+        # FIXME: Maybe this isn't to place to emit an error, especially when
+        # we cycle over multiple binarier to find the correct one.
+        LOG.error("GCC binary found is too old at "
```

Comment:
Not having gcc 13 on your machine and using codechecker is definitely not an error.


Now, the error message is like this
```
> CodeChecker analyzers
GCC binary found is too old at v7.5.0; minimum version is 13.0.0.
[ERROR 2023-10-11 13:49] - GCC binary found is too old at v7.5.0; minimum version is 13.0.0.
 clang-tidy
 clangsa
```

I would prefer something like this.
> CodeChecker analyzers
gcc /usr/bin/gcc GCC binary found is too old at v7.5.0; minimum version is 13.0.0. 
clang-tidy /local/workspace/llvm-project/build/bin/clang-tidy
clangsa /local/workspace/llvm-project/build/bin/clang-16

This way the user directly sees which analyzer binary is used.
Don't print the clang-tidy/clang binary location in this patch if you don't want, but it would look definitely better. :)

btw. where is cppcheck?